### DISCUSSION TITLE
GW1000-XS not connecting

### DISCUSSION THREAD

**Discussion by vanbuiten** [mletenay/home-assistant-goodwe-inverter#107] [Q&A] [answered]
Hello all!

First of all I have setup a solar roof on my house and my garage last year. I wanted to get more insight about my setup so I stumbled upon home assistant. I found this great repository and added it to home assistant. And it looks really nice and has great features getting the information from my inverter(s). Currently I am running HA in a docker container to learn it a bit more, I have a raspberry pi 4 that I want to setup.

I have no problem connecting to one of my inverters, the GW2000-XS.  However I cannot get the GW1000-XS inverter to connect. In the home assistant log I see the following:

```
2022-01-28 13:48:38 ERROR (MainThread) [custom_components.goodwe.config_flow] Connection error during GoodWe config flow: Unable to connect to the inverter at host=***.***.***.***, or your inverter is not supported yet.

Failures=[MaxRetriesException(), RequestFailedException('No valid response received even after 5 retries', 1), RequestFailedException('No valid response received even after 5 retries', 1), RequestFailedException('No valid response received even after 5 retries', 1)]
```

Does anyone else have this issue? And if so is there a solution for this?

**Comment by mletenay** [comment, marked as answer]
This is very likely problem of different version of ARM firmware on the smaller inverter, see readme file.
You need to update it to be able to communicate via local network.
(Good indicator is usually the mobile application (SolarGo) -  if you can connect to inverter via app, the HA should work as well.)

**Reply by vanbuiten** [reply]
Thank you for the answer. It is indeed a firmware issue, I will need to check who I need to contact to upgrade the inverter. I tried the SolarGo app and it was able to connect to the GW2000-XS and NOT to the GW1000-XS. For now I am getting the information I need by scraping the SEMS portal.
